### PR TITLE
WIP - expanding the new JDL grammar implementation.

### PR DESCRIPTION
**Pull request by bd82**

### CODE REVIEW

File: lib/dsl/poc/parser.js [hunk 1]
```diff
@@ -25,7 +25,12 @@ class JDLParser extends Parser {
           { ALT: () => $.SUBRULE($.entityDeclaration) },
           { ALT: () => $.SUBRULE($.relationDecl) },
           { ALT: () => $.SUBRULE($.enumDecl) },
-          { ALT: () => $.SUBRULE($.dtoDecl) }
+          { ALT: () => $.SUBRULE($.dtoDecl) },
+          { ALT: () => $.SUBRULE($.pagiDecl) },
```
Comment (MathieuAA) [inline lib/dsl/poc/parser.js]: can you, as much as possible, name the declaration so that `pagiDecl` is `paginationDeclaration` ?

Comment (bd82) [reply lib/dsl/poc/parser.js]: Sure, just copying from the existing grammar.

Comment (bd82) [reply lib/dsl/poc/parser.js]: maybe you meant **paginationDecl** ? because all the other rule names have a **decl** suffix?

**Comment by bd82** [comment]
When merging this it is probably best to do a squash before.

File: lib/dsl/poc/parser.js (hunk 1 above)
Comment (MathieuAA) [reply lib/dsl/poc/parser.js]: There is `entityDeclaration`, and I'd like to have everything nicely written like that. It's not important and another commit can be made to have that.

File: lib/dsl/poc/ast_builder.js [hunk 2]
```diff
@@ -19,10 +19,12 @@ const BaseJDLCSTVisitor = new JDLParser().getBaseCstVisitorConstructor();
  * 1. ctx.NAME[0].image --> ctx.NAME[0] ? ctx.NAME[0].image : "???"
  */
 class JDLAstBuilderVisitor extends BaseJDLCSTVisitor {
-  constructor() {
-    super();
-    this.validateVisitor();
-  }
+  // constructor() {
```
Comment (MathieuAA) [inline lib/dsl/poc/ast_builder.js]: Why is it not needed anymore?

File: lib/dsl/poc/lexer.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ function createToken(config) {
 }
 
 createToken({
-  name: 'WhiteSpace',
+  name: 'WHITESPACE',
```
Comment (MathieuAA) [inline lib/dsl/poc/lexer.js]: Nice =)

File: lib/dsl/poc/parser.js [hunk 4]
```diff
@@ -115,6 +128,198 @@ class JDLParser extends Parser {
...
+      $.CONSUME(t.FILTER);
+      $.SUBRULE($.negativeForDecl);
+      $.OPTION(() => {
+        $.SUBRULE($.exclusion);
+      });
+    });
+
+    // merged "subNoServerDecl", "subNoFluentMethod", "subFilterDecl" and "subNoClientDecl"
+    // as they are identical
+    // TODO: find a better name for this? should we call this FILTER_DEF?
```
Comment (MathieuAA) [inline lib/dsl/poc/parser.js]: Yes, good idea

File: lib/dsl/poc/ast_builder.js (hunk 2 above)
Comment (bd82) [reply lib/dsl/poc/ast_builder.js]: It is needed, I think I will re-add it and add the missing methods.

**Pull request merged**